src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    uint8 id = 0;
+
+    if (const CCharEntity* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        if (PChar->PParty)
+        {
+            if (PChar->PParty->m_PAlliance)
+                id = PChar->PParty->m_PAlliance->m_AllianceID;
```

Comment:
so this logic should stop and return here. If you have an alliance, push the id and gtfo. If you aren't in an alliance. push nil and move on.

At the very least, I don't necessarily have an issue with returning party ID if alliance doesn't exist, but pushing the character ID is weird as hell. It feels like you're coding this function to suit a specific purpose in a script, but you need to think outside of that use case.

Reply:
Ok no worries, ill take it out

Reply:
you know what, im going to split this into seperate functions, one for party one for alliance

Reply:
Would it be ok to make a specific function to do what i was doing before?, i would like to have a getCasketOwnerID that basicly did what i did in the first place

Reply:
I'll PR that seperately

Reply:
honestly, it doesn't seem that weird to me - an allianceid is just the partyid of the lead party, and a partyid is just the charid of the leader.  at the end of the day, anyone seeking an allianceid is just going to call partyid if its nil and then charid if thats nil too.  (and hopefully nobody is checking if an alliance exists by seeing if it has an id or not)